### ISSUE TITLE
100% CPU load with game locked up for (at least) hours

### ISSUE THREAD

**Issue by russell-taylor** [dungeons-of-moria/umoria#40]
Every once in awhile, the CPU load goes to 100% and the game stays locked up for at least hours.  Had previously encountered a brief version of this followed by a report of monsters being optimized.  I presume that was caused by some sort of replicating flea or louse causing too many of itself and needing to be cleaned up.  I don't know that this is the same cause as the essentially infinite lock, up but it seems likely.

This has happened fairly deep in the dungeon for me, so maybe it is some creature that shows up at larger depths.

This is on the mac, if it matters.  I'll try to recompile in debug mode and attach a debugger the next time it happens.

**Comment by mrcook** [comment]
Hi Russell, thanks the information on this. Am I right in assuming the optimization message was "`Compacting monsters...`"?

I took a quick look at that [compactMonsters()](https://github.com/dungeons-of-moria/umoria/blob/master/src/monster_manager.cpp#L253) function, but nothing stood out that could be causing this, so this will take some more research.

I don't see why this would be a mac only issue.

If you happen to have a saved game at a point just before this happens, and you're willing to share, that would be a huge help. No problem if not though.

**Comment by russell-taylor** [comment]
It is correct that the message I was referring to was "Compacting monsters...".  That message does not come up during the hangs, but it also took awhile to come up when it did happen which is why I thought it might be the same thing.  Could be a red herring.

**Comment by mrcook** [comment]
Hi Russell, I did some testing on this and was not able to trigger the issue you had.

I must of had over 50 "Compacting monsters..." messages, but there was never any slow down or freezing.

To achieve this I entered Wizard Mode (`CTRL-W`), found a nice big room a couple of levels down and generated a bunch of monsters (`&`) until I had some worm mass (`w`). Then just continually rest in a safe corridor (it helps to up your stats before doing this) while they multiplied. Slowly moving around the level and killing a few here and there seemed to speed up the number of comapcting messages.

Next time I get a chance I'll dig out my old mac and try on that system too.
